theia-electron/scripts/theia-electron-main.js

```diff
@@ -0,0 +1,13 @@
+const path = require('path')
+const os = require('os')
+
+// Use a set of builtin plugins in our application.
+process.env.THEIA_DEFAULT_PLUGINS = `local-dir:${path.resolve(__dirname, '..', 'plugins')}`
+
+// Lookup inside the user's home folder for more plugins, and accept user-defined paths.
+process.env.THEIA_PLUGINS = [
+    process.env.THEIA_PLUGINS, `local-dir:${path.resolve(os.homedir(), '.theia', 'plugins')}`,
+].filter(Boolean).join(',')
```

Comment:
@akosyakov IIRC when I tested this I was installing unpacked extensions there without any issues.